packages/plugins/websocket/lib/Socket.js

```diff
@@ -20,14 +20,17 @@ class Socket extends EventEmitter {
     }
     const messageHandler = ({data}) => this.emit('data', data)
     const errorHandler = err => {
-      this.emit('error', err)
+      this.emit(
+        'error',
+        err instanceof Error ? err : new Error(`connection error ${url}`)
+      )
     }
-    const closeHandler = () => {
+    const closeHandler = (code, reason) => {
       removeListener('open', openHandler)
       removeListener('message', messageHandler)
       removeListener('error', errorHandler)
       removeListener('close', closeHandler)
-      this.emit('close')
+      this.emit('close', reason ? {code, reason} : code)
```

Comment:
@smokku I checked, browsers never gives error for websocket issue.

evt.reason is a string sent by the server in the case of a clean close

Reply:
Unfortunately it is by design. There are security concerns related to passing websocket error messages up to application level.